<?xml version="1.0"?>
<launch>
    <arg name="use_rviz" default="$(optenv RACECAR_CLIENT false)" />
    <arg name="is_stage" default="false" />

    <group unless="$(arg is_stage)">
        <include file="$(find racecar_control)/launch/speed_controller.launch" />
    </group>

    <!-- Run Odometry -->
    <include file="$(find racecar_navigation)/launch/odom.launch"/>

    <node pkg="racecar_navigation" type="hound_dog" name="hound_dog" output="screen">
        <rosparam command="load" file="$(find racecar_navigation)/config/hound_line.yaml" ns="hound_line"/>

        <param name="speed_p" value="1.5" />
        <param name="error_p" value="0.6" />
        <param name="follow_distance" value="0.5" />
        <param name="max_distance" value="2.5" />
    </node>

    <!-- Rviz -->
    <include file="$(find racecar_core)/launch/rviz.launch" if="$(arg use_rviz)">
        <arg name="config" value="hound"/>
    </include>
</launch>
